### PR TITLE
Add new toolbar/debugger integration

### PR DESCRIPTION
**Pull request by xepozz**
image
image

### CODE REVIEW

**Review by rustamwin — COMMENTED** [review]
What about add a new middleware & asset instead of replacing the current one?

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
The code itself is OK but...

Should it be done here? Because I assume after it's done, https://github.com/yiisoft/yii-debug-viewer/tree/master/src/Panels would be unused code, right?

File: resources/views/debug.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+$this->addJsVars($assetManager->getJsVars());
+
+$this->beginPage();
+?>
+    <!doctype html>
+    <html lang="en">
+    <head>
+        <meta charset="utf-8"/>
+        <meta name="viewport" content="width=device-width,initial-scale=1"/>
+        <meta name="description" content="Aggregator for Yii3 specific frontends"/>
```
Comment (samdark) [inline resources/views/debug.php:27]: What's meant by "Aggregator" here? I guess the description should say something else.

File: resources/views/debug.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+              href="https://yiisoft.github.io/yii-dev-panel/android-chrome-512x512.png"/>
+        <meta name="apple-mobile-web-app-capable" content="yes">
+        <meta name="apple-mobile-web-app-title" content="Yii Dev Panel">
+        <meta name="application-name" content="Yii Dev Panel">
+        <meta name="msapplication-TileColor" content="#01617b">
+        <meta name="theme-color" content="#01617b">
+        <link rel="manifest" href="https://yiisoft.github.io/yii-dev-panel/manifest.json"/>
+        <title>Yii Dev Panel</title>
+        <?php
+        $this->head() ?>
```
Comment (samdark) [inline resources/views/debug.php:46]:
```suggestion
        <?php $this->head() ?>
```

File: resources/views/debug.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    </head>
+    <body style="display:flex;flex-direction:column;min-height:100vh;justify-content:space-between">
+    <?php
+    $this->beginBody(); ?>
```
Comment (samdark) [inline resources/views/debug.php:50]:
```suggestion
    $this->beginBody() ?>
```

**Review by samdark — APPROVED** [review]

**Review by samdark — APPROVED** [review]

File: src/Middleware/DevPanelMiddleware.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+     */
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        $this->assetManager->register(DevPanelAsset::class);
+        $this->view->registerJs(
+            <<<JS
+            const devPanelContainerId = 'yii-dev-panel';
+            const devPanel = document.createElement('div');
+            devPanel.setAttribute('id', devPanelContainerId);
+            devPanel.style.flex= "1";
```
Comment (samdark) [inline src/Middleware/DevPanelMiddleware.php:35]:
```suggestion
            devPanel.style.flex = "1";
```

File: src/Asset/DevPanelAsset.php [hunk 5]
```diff
@@ -13,17 +13,17 @@ final class DevPanelAsset extends AssetBundle
...
-            //'http://localhost:4173/bundle.js',
+            //'https://yiisoft.github.io/yii-dev-panel/bundle.js',
+            'http://localhost:4173/bundle.js',
             WebView::POSITION_END,
             'type' => 'module',
         ],
-        'https://yiisoft.github.io/yii-dev-panel/registerSW.js',
-        //'http://localhost:4173/registerSW.js',
+        //'https://yiisoft.github.io/yii-dev-panel/registerSW.js',
+        'http://localhost:4173/registerSW.js',
```
Comment (samdark) [inline src/Asset/DevPanelAsset.php:22]: Is that teporary?

**Pull request merged**